server/src/main/java/org/candlepin/dto/manifest/v1/PoolTranslator.java

```diff
@@ -150,8 +162,47 @@ public PoolDTO populate(ModelTranslator modelTranslator, Pool source, PoolDTO de
             else {
                 dest.setDerivedProvidedProducts(Collections.emptySet());
             }
+
+            // accumulate provided products for stacked entitlements
+            if (PoolType.STACK_DERIVED.equals(source.getType())) {
+                Collection<Product> providedProducts = accumulateStackDerivedPoolProvidedProducts(source);
+                if (source.getDerivedProduct() != null) {
+                    for (Product providedProduct : providedProducts) {
```

Comment:
These need to be n-tier aware translations

Reply:
Done.